pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -58,6 +58,15 @@ class PublishOptions(object):
     :meth:`~pubtools.pulplib.Distributor.is_rsync` is ``True``.
     """
 
+    rsync_extra_args = pulp_attrib(
+        default=attr.Factory(frozenlist), type=list, converter=frozenlist
```

Comment:
The default change I think should be left out, and default=None was better.

The problem with using an empty list here as the default is that it's ambiguous whether empty list means "I don't want to set any rsync_extra_args", or means "I want rsync_extra_args to be set to an empty list".  It should be left as default=None so we can support all these cases:

- None: do not override rsync_extra_args at all, just let the pulp server do what it wants.
- empty list: set rsync_extra_args to an empty list (effectively disabling whatever was the default config in the pulp server).
- non-empty list: set rsync_extra_args to the list I specify (overriding the config in the pulp server).

Reply:
When default=None, the following error is generated;

```
ImportError while loading conftest '/home/ngillett/src/pubtools-pulplib/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from pubtools.pulplib import Client
pubtools/pulplib/__init__.py:1: in <module>
    from ._impl.client import Client, PulpException, TaskFailedException
pubtools/pulplib/_impl/client/__init__.py:1: in <module>
    from .client import Client
pubtools/pulplib/_impl/client/client.py:14: in <module>
    from ..page import Page
pubtools/pulplib/_impl/page.py:10: in <module>
    from .model.attr import pulp_attrib
pubtools/pulplib/_impl/model/__init__.py:2: in <module>
    from .repository import (
pubtools/pulplib/_impl/model/repository/__init__.py:1: in <module>
    from .base import Repository, PublishOptions, SyncOptions
pubtools/pulplib/_impl/model/repository/base.py:132: in <module>
    class Repository(PulpObject, Deletable):
pubtools/pulplib/_impl/model/repository/base.py:366: in Repository
    def publish(self, options=PublishOptions()):
E   TypeError: 'NoneType' object is not iterable
ERROR: InvocationError for command /home/ngillett/src/pubtools-pulplib/.tox/py36/bin/pytest -v (exited with code 4)
```
This seems to be due to the validator attached by pulp_attrib() because...
```
>>> isinstance(None, list)
False
```
I noticed that there's no attr defined in repository.base with default=None so, thinking this error may be a factor, I followed suit.
Maybe this necessitates a change to pulp_attrib(), though...